app/templates/layouts/_base.html

```diff
@@ -46,27 +46,43 @@
 
     {% endblock styles -%}
     {% endblock head -%}
-    <!-- Google Analytics -->
-
-    {% if analytics_ua_id  %}
+    <!-- Google Tag Manager -->
+    {% if analytics_gtm_id and analytics_gtm_env_id %}
     <script nonce="{{ csp_nonce() }}">
-    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
-
-    ga('create', '{{analytics_ua_id}}', 'auto');
-    ga('set', 'anonymizeIp', true);
-    // Language is setup as a custom dimension in google analytics
-    ga('set', 'dimension1', '{{language_code}}' || 'en');
-    ga('send', 'pageview');
+      (function(w, d, s, l, i) {
+        w['GoogleTagManagerObject']=l;
+        w[l] = w[l] || [];
+        w[l].push({ "gtm.start": new Date().getTime(), event: "gtm.js" });
+        var f = d.getElementsByTagName(s)[0],
+          j = d.createElement(s),
+          dl = l != "dataLayer" ? "&l=" + l : "";
+        j.async = true;
+        j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl + "&gtm_auth={{analytics_gtm_id}}&gtm_preview=env-8&gtm_cookies_win=x";
+        f.parentNode.insertBefore(j, f);
+      })(window, document, "script", "dataLayer", "{{analytics_gtm_env_id}}");
     </script>
     {% endif %}
-    <!-- End Google Analytics -->
+    <!-- End Google Tag Manager -->
   </head>
 
   <body>
 
+    <!-- Google Tag Manager (noscript) -->
```

Comment:
This includes a `script` tag in the noscript part? huh?

Reply:
This is taken from the snippet from GTM that I had to change to allow me to pass in the environment variables and to check the environment variables exist before rendering the tags. Ill remove the "(noscript)" from the comment